pkg/kubelet/apis/podresources/server_v1.go

```diff
@@ -27,22 +27,24 @@ import (
 	"k8s.io/kubelet/pkg/apis/podresources/v1"
 )
 
-// podResourcesServerV1alpha1 implements PodResourcesListerServer
+// podResourcesServerV1 implements PodResourcesListerServer
 type v1PodResourcesServer struct {
 	podsProvider    PodsProvider
 	devicesProvider DevicesProvider
 	cpusProvider    CPUsProvider
 	memoryProvider  MemoryProvider
+	draProvider     DRAProvider
 }
 
 // NewV1PodResourcesServer returns a PodResourcesListerServer which lists pods provided by the PodsProvider
 // with device information provided by the DevicesProvider
-func NewV1PodResourcesServer(podsProvider PodsProvider, devicesProvider DevicesProvider, cpusProvider CPUsProvider, memoryProvider MemoryProvider) v1.PodResourcesListerServer {
+func NewV1PodResourcesServer(podsProvider PodsProvider, devicesProvider DevicesProvider, cpusProvider CPUsProvider, memoryProvider MemoryProvider, draProvider DRAProvider) v1.PodResourcesListerServer {
```

Comment:
Can this use the new `PodResourcesProviders` type introduced in a previous commit?

Reply:
This code is cherry-pick [1] from node: podresources: add rate limiting. see [2]. 
I am not sure why @ffromani did pass it also to  NewV1PodResourcesServer.


[1] https://github.com/kubernetes/kubernetes/pull/115852/commits/f81e542f6411ddea2a10e21a85aabc6552b2b362
[2] https://github.com/kubernetes/kubernetes/pull/115852/commits/f81e542f6411ddea2a10e21a85aabc6552b2b362#r1112351927

Reply:
Just because the scope of my change was limited :) I think @klueska has a good point here, feel free to change this commit in this PR as you see fit (my PR should not delay yours), I'll happily rebase mine later.